translib/transformer/xfmr_intf.go

```diff
@@ -1072,9 +1102,17 @@ var DbToYang_intf_eth_auto_neg_xfmr FieldXfmrDbtoYang = func(inParams XfmrParams
 	}
 	intTbl := IntfTypeTblMap[intfType]
 
-	tblName, _ := getPortTableNameByDBId(intTbl, inParams.curDb)
-	pTbl := data[tblName]
-	prtInst := pTbl[inParams.key]
+	// https://github.com/sonic-net/sonic-buildimage/issues/9595
+	tblName, _ := getPortTableNameByDBId(intTbl, db.ConfigDB)
+	d := inParams.dbs[db.ConfigDB]
+	pTbl := db.TableSpec{Name: tblName}
+
+	prtInst, tblErr := d.GetEntry(&pTbl, db.Key{Comp: []string{inParams.key}})
+	if tblErr != nil {
+		log.Info("DbToYang_intf_eth_auto_neg_xfmr key not found : ", inParams.key)
+		return result, errors.New("Interface not found : " + inParams.key)
```

Comment:
Use tlerr.NotFound error. In general avoid the generic errors.New as much as possible -- REST & gNMI servers cannot map it to appropriate protocol error codes. Use the specific error types defined in the tlerr package.